src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.TargetSymbolResolver.cs

```diff
@@ -5,14 +5,16 @@
 using System.Collections.Immutable;
 using System.Diagnostics;
 using System.Linq;
+using System.Runtime.InteropServices;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Diagnostics
 {
     internal partial class SuppressMessageAttributeState
     {
-        private static string s_suppressionPrefix = "~";
+        private const string s_suppressionPrefix = "~";
 
+        [StructLayout(LayoutKind.Auto)]
```

Comment:
By default structs, unlike classes, use `[StructLayout(LayoutKind.Sequntial)]` that guarantees the order of fields in interop scenarios. For structs not indended for interop, specifying `[StructLayout(LayoutKind.Auto)]` allows CLR to reorder and realign fields, that may cause some perf improvements.